cli/src/commands/bookmark/forget.rs

```diff
@@ -57,22 +64,46 @@ pub fn cmd_bookmark_forget(
     let repo = workspace_command.repo().clone();
     let matched_bookmarks = find_forgettable_bookmarks(repo.view(), &args.names)?;
     let mut tx = workspace_command.start_transaction();
+    let mut forgotten: usize = 0;
+    let mut untracked: usize = 0;
     for (name, bookmark_target) in &matched_bookmarks {
-        tx.repo_mut()
-            .set_local_bookmark_target(name, RefTarget::absent());
-        for (remote_name, _) in &bookmark_target.remote_refs {
+        if bookmark_target.local_target.is_present() {
             tx.repo_mut()
-                .set_remote_bookmark(name, remote_name, RemoteRef::absent());
+                .set_local_bookmark_target(name, RefTarget::absent());
+            forgotten += 1;
+        }
+        for (remote_name, remote_target) in &bookmark_target.remote_refs {
+            // If `--include-remotes` is specified, we forget the corresponding remote
+            // bookmarks instead of untracking them
+            if args.include_remotes {
+                tx.repo_mut()
+                    .set_remote_bookmark(name, remote_name, RemoteRef::absent());
+                forgotten += 1;
+                continue;
+            }
+            // Git-tracking remote bookmarks cannot be untracked currently, so skip them
+            if jj_lib::git::is_special_git_remote(remote_name) {
+                continue;
+            }
+            if remote_target.is_tracking() {
+                tx.repo_mut().untrack_remote_bookmark(name, remote_name);
+                untracked += 1;
+            }
         }
     }
-    writeln!(ui.status(), "Forgot {} bookmarks.", matched_bookmarks.len())?;
-    tx.finish(
-        ui,
-        format!(
-            "forget bookmark {}",
-            matched_bookmarks.iter().map(|(name, _)| name).join(", ")
-        ),
-    )?;
+    if untracked == 0 {
+        writeln!(ui.status(), "Forgot {forgotten} bookmarks.")?;
+    } else {
+        writeln!(
+            ui.status(),
+            "Forgot {forgotten} bookmarks and untracked {untracked} remote bookmarks."
+        )?;
+    }
```

Comment:
nit: simply print these as separate messages?

It might be also better to print "N local bookmarks" and "M remote bookmarks" (instead of forgotten + untracked) as we kind of define "forget local bookmarks" is the operation to delete + untrack.